### PR TITLE
Fix issue #10806

### PR DESCRIPTION
**Pull request by Upabjojr**
#10806

### CODE REVIEW

**Comment by Upabjojr** [comment]
Better try a different approach: _Integral_ shall have the same precedence as _Atom_ (as it currently is in master) except for the LaTeX printer, where it shall have the same precedence as _Mul_.

**Comment by Upabjojr** [comment]
@gxyd what do you think?

**Comment by gxyd** [comment]
One quick comment: issue with `Limit` is still there.
```
>>> print(latex(Limit(x, x, 1)**2))
\lim_{x \to 1^+} x^{2}
```
I will look into it.

**Comment by Upabjojr** [comment]
> One quick comment: issue with Limit is still there.

That's an easy fix. Other atomic expressions that have a multiplication-like precedence in LaTeX?

**Comment by Upabjojr** [comment]
Also _Derivative_, _Sum_, _Product_ are affected.

File: sympy/printing/latex.py [hunk 1]
```diff
@@ -144,8 +144,18 @@ def __init__(self, settings=None):
 
         self._delim_dict = {'(': ')', '[': ']'}
 
-    def parenthesize(self, item, level):
-        if precedence(item) <= level:
+    def parenthesize(self, item, level, strict=False):
+        # Integral has the precedence of Mul in LaTeX, the precedence of Atom
+        # for other printers:
+        from sympy import Integral, Sum, Product, Derivative
```
Comment (gxyd) [inline sympy/printing/latex.py]: Here `Limit` can also be added.

File: sympy/printing/latex.py [hunk 2]
```diff
@@ -144,8 +144,18 @@ def __init__(self, settings=None):
 
         self._delim_dict = {'(': ')', '[': ']'}
 
-    def parenthesize(self, item, level):
-        if precedence(item) <= level:
+    def parenthesize(self, item, level, strict=False):
+        # Integral has the precedence of Mul in LaTeX, the precedence of Atom
```
Comment (gxyd) [inline sympy/printing/latex.py]: Please change the comment to reflect for `Sum`, `Product`, `Limit` as well.

**Comment by gxyd** [comment]
I like the approach of the PR. Everything looks fine, except one issue i have. In "terminal"
```
>>> from sympy import *
>>> x, y, z = symbols('x y z')
>>> init_printing()
>>> Limit(Limit(x**2, x, y)**(S(5)/2), y, 2)
            5/2
           2   
 lim  lim x    
y─→2⁺x─→y⁺     

>>> Limit(Limit(x**2, x, y), y, 2)**(S(5)/2)
            5/2
           2   
 lim  lim x    
y─→2⁺x─→y⁺     
```
No brackets are put around `Limit of x**2` (in first case), and no brackets around whole `Limit` object.
While this issue is *not* there in Jupyter notebook.

**Comment by gxyd** [comment]
+1. Probably that is unrelated to the PR. Will merge in 12 hrs.

**Comment by Upabjojr** [comment]
> Everything looks fine, except one issue i have. In "terminal"

Maybe it's an issue in the unicode printer?

**Comment by Upabjojr** [comment]
Sorry, in the _pprint_, not in the unicode printer.

**Comment by Upabjojr** [comment]
@gxyd I have a fix for your issue. Will post a new PR as soon as this one is merged.

**Comment by gxyd** [comment]
> Sorry, in the pprint, not in the unicode printer.

Yes, that is a issue in `pprint`.
> I have a fix for your issue. Will post a new PR as soon as this one is merged.

Okay, that will be good.

**Comment by Upabjojr** [comment]
https://github.com/sympy/sympy/pull/10940

**Comment by gxyd** [comment]
Thanks @Upabjojr merging this.
You may also like to express your views [for the question](https://groups.google.com/forum/#!topic/sympy/WiTJxz2tVyI), i asked on google groups.

**Pull request merged**